File: Modules/Lookups/Models/LookupValue.php
<?php

namespace Modules\Lookups\Models;

use App\Support\Traits\Models\HasTranslation;
use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\BelongsTo;

class LookupValue extends Model
{
    use HasTranslation;

    protected $fillable = [
        'lookup_type_id',
        'value_key',
        'label_ar',
        'label_en',
        'sort_order',
        'is_active',
    ];

    protected $casts = ['is_active' => 'boolean'];

    public function type(): BelongsTo
    {
        return $this->belongsTo(LookupType::class, 'lookup_type_id');
    }

    protected function getTranslationColumns(): array
    {
        return ['label_ar', 'label_en'];
    }
}

File: Solid.Api/Database/Entities/LookupValue.cs
namespace Solid.Api.Database.Entities;

public sealed class LookupValue
{
    public long Id { get; set; }

    public long LookupTypeId { get; set; }

    public string ValueKey { get; set; } = string.Empty;

    public string LabelAr { get; set; } = string.Empty;

    public string LabelEn { get; set; } = string.Empty;

    public byte SortOrder { get; set; }

    public bool IsActive { get; set; } = true;

    public DateTime? CreatedAt { get; set; }

    public DateTime? UpdatedAt { get; set; }

    public LookupType LookupType { get; set; } = null!;

    public ICollection<AddictionProfile> AddictionDurationProfiles { get; set; } = [];

    public ICollection<AddictionProfile> EducationLevelProfiles { get; set; } = [];
}
